reo/views.py

```diff
@@ -758,3 +752,64 @@ def absorption_chiller_defaults(request):
                                                                             tb.format_tb(exc_traceback))
         log.debug(debug_msg)
         return JsonResponse({"Error": "Unexpected error in absorption_chiller_defaults endpoint. Check log for more."}, status=500)
+
+def schedule_stats(request):
+    """
+    Get a summary of a yearly profile by calculating the weekday, weekend, and total hours by month (e.g. for chp_unavailability_periods viewing in the UI)
+    :param year: required input year for establishing the calendar
+    :param chp_prime_mover: required if chp_unavailability_periods is not provided, otherwise not required or used
+    :param chp_unavailability_periods: list of dictionaries, one dict per unavailability period (as defined in nested_inputs.py)
+    :return weekday_weekend_total_hours_by_month: nested dictionary with 12 keys (one for each month) each being a dictionary of weekday_hours, weekend_hours, and total_hours
+    """
+    try:
+        if request.method == "GET":
+            if not (request.GET["year"] and request.GET["chp_prime_mover"]):
+                ValueError("A GET request method is only applicable for getting the default stats using year and chp_prime_mover as query params")
+            year = int(request.GET["year"])
+            chp_prime_mover = request.GET["chp_prime_mover"]
+            chp_unavailability_periods = None
+        elif request.method == "POST":
+            request_dict = json.loads(request.body)
+            year = int(request_dict.get('year'))
+            chp_prime_mover = request_dict.get('chp_prime_mover')
+            chp_unavailability_periods = request_dict.get('chp_unavailability_periods')
```

Comment:
You might need to handle lists a little differently I think Andrew's code is set up to give each list item a unique variable id see https://github.com/NREL/REopt_Lite_API/blob/c2dbc2045c2d1b2ce950a19b01ce63f5adc901c5/reo/views.py#L299

In that code you can see we accept a list [] or separate variable names doe_reference_name[0]=, doe_reference_name[1]=

Reply:
Ok, I'm going to check with Andrew on this. I'm wondering if it's different because of how I'm reading the POST request in differently. It's working locally for me with a POST (i.e. reading in lists).

Reply:
Yup, it will work either way, for the simulated loads endpoint anyways Andrew didn't seem to have much control over how arrays were handled